src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/consumer/impl/MainConsumerImpl.java

```diff
@@ -36,26 +39,42 @@ void init() {
     @Override
     @KafkaListener(topics = "${kafka.main.topic}", containerFactory = "kafkaListenerContainerFactory", groupId = "main-group")
     public void readAndProcessMainTopic(@Payload ChipsRestInterfacesSend data,
-                                        @Headers MessageHeaders headers){
-        processMessage("main-consumer", data, headers);
+                                        @Header(KafkaHeaders.OFFSET) Long offset,
+                                        @Header(KafkaHeaders.RECEIVED_PARTITION_ID) Integer partition,
+                                        @Header(KafkaHeaders.GROUP_ID) String groupId
+    ){
+        processMessage(groupId, data, offset, partition);
     }
 
     @Override
     @KafkaListener(topics = "${kafka.retry.topic}", containerFactory = "kafkaRetryListenerContainerFactory", groupId = "retry-group")
-    public void readAndProcessRetryTopic(@Payload ChipsRestInterfacesSend data,
-                                        @Headers MessageHeaders headers){
-        processMessage("retry-consumer", data, headers);
-    }
+    public void readAndProcessRetryTopic(@Payload List<ChipsRestInterfacesSend> messages,
+                                         @Header(KafkaHeaders.OFFSET) List<Long> offsets,
+                                         @Header(KafkaHeaders.RECEIVED_PARTITION_ID) List<Integer> partitions,
+                                         @Header(KafkaHeaders.GROUP_ID) String groupId
+    ){
+
+        logger.debug(String.format("%s, received %s messages", groupId, messages.size()));
```

Comment:
The log and the for loop could go into a processMessages method.

Reply:
I'm not sure what that would achieve as the log and for loop is the only thins this method does, and this is the only place in the code base this code appears